sycl/include/CL/__spirv/spirv_ops.hpp

```diff
@@ -136,100 +160,372 @@ __SPIRV_ATOMICS(__SPIRV_ATOMIC_UNSIGNED, unsigned long long)
+extern SYCL_EXTERNAL void
+__spirv_ReadPipeBlockingINTEL(RPipeTy<dataT> Pipe, dataT *Data, int32_t Size,
+                              int32_t Alignment) noexcept;
 template <typename dataT>
-extern void __spirv_WritePipeBlockingINTEL(WPipeTy<dataT> Pipe,
-                                           const dataT *Data, int32_t Size,
-                                           int32_t Alignment) noexcept;
+extern SYCL_EXTERNAL void
+__spirv_WritePipeBlockingINTEL(WPipeTy<dataT> Pipe, const dataT *Data,
+                               int32_t Size, int32_t Alignment) noexcept;
 template <typename dataT>
-extern RPipeTy<dataT> __spirv_CreatePipeFromPipeStorage_read(
+extern SYCL_EXTERNAL RPipeTy<dataT> __spirv_CreatePipeFromPipeStorage_read(
     const ConstantPipeStorage *Storage) noexcept;
 template <typename dataT>
-extern WPipeTy<dataT> __spirv_CreatePipeFromPipeStorage_write(
+extern SYCL_EXTERNAL WPipeTy<dataT> __spirv_CreatePipeFromPipeStorage_write(
     const ConstantPipeStorage *Storage) noexcept;
 
-extern void __spirv_ocl_prefetch(const __attribute__((opencl_global)) char *Ptr,
-                                 size_t NumBytes) noexcept;
+extern SYCL_EXTERNAL void
+__spirv_ocl_prefetch(const __attribute__((opencl_global)) char *Ptr,
+                     size_t NumBytes) noexcept;
+
+SYCL_EXTERNAL
+int __spirv_ocl_printf(const __attribute__((opencl_constant)) char *fmt, ...);
+
+#define COMPARISON(Order, Cmp)                                                 \
+  template <typename ReTTT, typename T1, typename T2>                          \
+  extern SYCL_EXTERNAL                                                         \
+  ReTTT __spirv_F##Order##Cmp(T1, T2);
+
+#define ALL_COMPARISON(Order)                                                  \
+  COMPARISON(Order, Equal)                                                     \
+  COMPARISON(Order, NotEqual)                                                  \
+  COMPARISON(Order, LessThan)                                                  \
+  COMPARISON(Order, GreaterThan)                                               \
+  COMPARISON(Order, LessThanEqual)                                             \
+  COMPARISON(Order, GreaterThanEqual)
+
+ALL_COMPARISON(Unord)
+ALL_COMPARISON(Ord)
+
+#undef COMPARISON
+#undef ALL_COMPARISON
+
+#define COMPARISON(Cmp)                                                        \
+  template <typename ReTTT, typename T1, typename T2>                          \
+  extern SYCL_EXTERNAL                                                         \
+  ReTTT __spirv_##Cmp(T1, T2);
+
+COMPARISON(IEqual)
+COMPARISON(INotEqual)
+
+COMPARISON(ULessThan)
+COMPARISON(UGreaterThanEqual)
+COMPARISON(ULessThanEqual)
+COMPARISON(UGreaterThan)
+
+COMPARISON(SLessThan)
+COMPARISON(SGreaterThanEqual)
+COMPARISON(SLessThanEqual)
+COMPARISON(SGreaterThan)
+
+COMPARISON(LessOrGreater)
+
+#undef COMPARISON
+
+template <typename ReTTT, typename T1>
```

Comment:
```suggestion
template <typename ReTTT, typename T>
```
What is the meaning of 1 in the typename? Copy-paste?
Please, remove 1 from all the declarations using a single type.